cmd/restic/cmd_mount.go

```diff
@@ -0,0 +1,301 @@
+// +build !openbsd
+
+package main
+
+import (
+	"encoding/binary"
+	"fmt"
+	"os"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/restic/restic"
+	"github.com/restic/restic/backend"
+	"github.com/restic/restic/crypto"
+	"github.com/restic/restic/pack"
+	"github.com/restic/restic/repository"
+
+	"bazil.org/fuse"
+	"bazil.org/fuse/fs"
+)
+
+type CmdMount struct {
+	global *GlobalOptions
+	ready  chan struct{}
+}
+
+func init() {
+	_, err := parser.AddCommand("mount",
+		"mount a repository",
+		"The mount command mounts a repository read-only to a given directory",
+		&CmdMount{
+			global: &globalOpts,
+			ready:  make(chan struct{}, 1),
+		})
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (cmd CmdMount) Usage() string {
+	return "MOUNTPOINT"
+}
+
+func (cmd CmdMount) Execute(args []string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("wrong number of parameters, Usage: %s", cmd.Usage())
+	}
+
+	repo, err := cmd.global.OpenRepository()
+	if err != nil {
+		return err
+	}
+
+	err = repo.LoadIndex()
+	if err != nil {
+		return err
+	}
+
+	mountpoint := args[0]
+	if _, err := os.Stat(mountpoint); err != nil {
+		if os.IsNotExist(err) {
+			cmd.global.Verbosef("Mountpoint %s doesn't exist, creating it\n", mountpoint)
+			err = os.Mkdir(mountpoint, os.ModeDir|0700)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	c, err := fuse.Mount(
+		mountpoint,
+		fuse.ReadOnly(),
+		fuse.FSName("restic"),
```

Comment:
It's alright to have multiple file systems with the same name, here's the output of my `df -a`:

```
proc                       0         0          0    - /proc
sys                        0         0          0    - /sys
dev                  1529256         0    1529256   0% /dev
run                  1531952       592    1531360   1% /run
/dev/sda4          144064184 131102472    5620608  96% /
securityfs                 0         0          0    - /sys/kernel/security
tmpfs                1531952       176    1531776   1% /dev/shm
devpts                     0         0          0    - /dev/pts
tmpfs                1531952         0    1531952   0% /sys/fs/cgroup
cgroup                     0         0          0    - /sys/fs/cgroup/systemd
pstore                     0         0          0    - /sys/fs/pstore
cgroup                     0         0          0    - /sys/fs/cgroup/net_cls
cgroup                     0         0          0    - /sys/fs/cgroup/freezer
cgroup                     0         0          0    - /sys/fs/cgroup/cpu,cpuacct
cgroup                     0         0          0    - /sys/fs/cgroup/devices
cgroup                     0         0          0    - /sys/fs/cgroup/blkio
cgroup                     0         0          0    - /sys/fs/cgroup/cpuset
cgroup                     0         0          0    - /sys/fs/cgroup/memory
systemd-1                  -         -          -    - /proc/sys/fs/binfmt_misc
hugetlbfs                  0         0          0    - /dev/hugepages
mqueue                     0         0          0    - /dev/mqueue
tmpfs                1531952    199780    1332172  14% /tmp
debugfs                    0         0          0    - /sys/kernel/debug
configfs                   0         0          0    - /sys/kernel/config
fusectl                    0         0          0    - /sys/fs/fuse/connections
tmpfs                 306392        12     306380   1% /run/user/1000
restic                     0         0          0    - /tmp/restic-mount
restic                     0         0          0    - /tmp/restic-mount2
binfmt_misc                0         0          0    - /proc/sys/fs/binfmt_misc
```

So, yeah, mounting the same repo on 2 different mount points (`/tmp/restic-mount` and `/tmp/restic-mount2` here) works.